sentry-opentelemetry/lib/sentry/opentelemetry/span_processor.rb

```diff
@@ -0,0 +1,159 @@
+require 'singleton'
+
+module Sentry
+  module OpenTelemetry
+    TraceData = Struct.new(:trace_id, :span_id, :parent_span_id, :parent_sampled, :baggage)
+
+    class SpanProcessor < ::OpenTelemetry::SDK::Trace::SpanProcessor
+      include Singleton
+
+      SEMANTIC_CONVENTIONS = ::OpenTelemetry::SemanticConventions::Trace
+
+      # The mapping from otel span ids to sentry spans
+      # @return [Hash]
+      attr_reader :span_map
+
+      def initialize
+        @span_map = {}
+      end
+
+      def on_start(otel_span, parent_context)
+        return unless Sentry.initialized? && Sentry.configuration.instrumenter == :otel
+        return unless otel_span.context.valid?
+        return if from_sentry_sdk?(otel_span)
+
+        trace_data = get_trace_data(otel_span, parent_context)
+
+        sentry_parent_span = @span_map[trace_data.parent_span_id] if trace_data.parent_span_id
```

Comment:
Still a little concerning having this hash shared between multiple-threads. Could 2 or more otel spans share the same `parent_span_id`?

Reply:
we're just getting the sentry span corresponding to the otel span, multiple spans can have the same parent and that shouldn't cause any problems.

Regarding concurrency, we can eventually wrap stuff in mutexes _if someone complains_ but given that ruby hashes work with the GIL in cruby at least, this is fine for now. We might need to use `ConcurrentHashMap` in jruby or other implementations, but I would not worry about this now.

This is an MVP, we're getting it out and seeing if there's interest at all in such features. We will iterate once there is proven demand from the community.